src/utils.js

```diff
@@ -224,6 +226,20 @@ function isDOMElement(element) {
   }
 }
 
+/**
+ * When built, this constant will signify if the current platform is React Native.
+ * This can be used to optimize codepaths for other platforms without conflating
+ * concepts because of the React Native target.
+ * @return {Boolean} whether or not the platform is React Native
+ */
+/* istanbul ignore next (react-native) */
+const isReactNative = preval`module.exports = process.env.BUILD_REACT_NATIVE === 'true'`
+
+// Disabling coverage here is necessary due to a weird deal with
+// babel-plugin-istanbul + preval.macro. No idea...
+/* istanbul ignore next (preact) */
+const isPreact = preval`module.exports = process.env.BUILD_PREACT === 'true'`
```

Comment:
I think that these will actually need to be inlined as they were before to make sure the code is removed properly. I realize it's not as clean, but it'll work better.

Reply:
This PR got me interested and I agree - better to have those branches DCE-ed.

Reply:
Thanks! I'm less concerned about the `umd` builds and more concerned about all the other ones. Could you run the builds and see if Rollup was able to eliminate those? If not, could you try to inline the preval usage and see if Rollup can eliminate it that way. I know it's not pretty, but we've gotta work with the tools we have.

Reply:
Ah, gotcha. Rollup does not eliminate them there. I'll add the `preval`s back in, inline!

Do you have more reading for me to learn more about why someone might not use the `umd.min` version of the build so I can understand the benefit of optimizing those builds?